site/docs/types/data_type_classes.md

```diff
@@ -44,18 +44,42 @@ Compound type classes are type classes that need to be configured by means of a
 
 ## User-Defined Types
 
-User-defined type classes can be created using a combination of pre-defined types. User-defined types are defined as part of [simple extensions](../extensions/index.md#simple-extensions). An extension can declare an arbitrary number of user defined extension types.
+User-defined type classes can be created using a combination of pre-defined types. User-defined types are defined as part of [simple extensions](../extensions/index.md#simple-extensions). An extension can declare an arbitrary number of user defined extension types. Once a type has been declared, it can be used in function declarations.
 
 A YAML example of an extension type is below:
 
 ```yaml
-  name: point
-  structure:
-    longitude: i32
-    latitude: i32
+name: point
+structure:
+  longitude: i32
+  latitude: i32
+```
+
+This declares a new type (namespaced to the associated YAML file) called "point". This type is composed of two `i32` values named longitude and latitude.
+
+### Structure and opaque types
+
+The name-type object notation used above is syntactic sugar for `NSTRUCT<longitude: i32, latitude: i32>`. The following means the same thing:
+
+```yaml
+name: point
+structure: "NSTRUCT<longitude: i32, latitude: i32>"
 ```
 
-This declares a new type (namespaced to the associated YAML file) called "point". This type is composed of two `i32` values named longitude and latitude. Once a type has been declared, it can be used in function declarations.  [TBD: should field references be allowed to dereference the components of a user defined type?]
+The structure field of a type is only intended to inform systems that don't have built-in support for the type how they can transfer the data type from one point to another without unnecessary serialization/deserialization *and* without loss of type safety. Note that it is currently not possible to "unpack" a user-defined type class into its structure type or components thereof using `FieldReference`s or any other specialized record expression; if support for this is desired for a particular type, this can be accomplished with an extension function.
```

Comment:
Why? Those are pretty unrelated to this. I very intentionally write "type *class*" rather than just "type" to make sure I'm not accidentally writing down something ambiguous.